client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodProxyDetails.java

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.rest.v2;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Details of a specific method from an interface that was generated from a Swagger specification.
+ * This is generally used by a proxy implementation of the Swagger interface.
+ */
+class SwaggerMethodProxyDetails {
+    private String method;
+    private String relativePath;
+    private final List<Substitution> hostSubstitutions = new ArrayList<>();
+    private final List<Substitution> pathSubstitutions = new ArrayList<>();
+    private final List<Substitution> querySubstitutions = new ArrayList<>();
+    private final List<Substitution> headerSubstitutions = new ArrayList<>();
+    private Integer bodyContentMethodParameterIndex;
+
+    /**
+     * Get the HTTP method that will be used to complete the Swagger method's request.
+     * @return The HTTP method that will be used to complete the Swagger method's request.
+     */
+    public String method() {
+        return method;
+    }
+
+    /**
+     * Set the HTTP method that will be used to complete the Swagger method's request.
+     * @param method The HTTP method that will be used to complete the Swagger method's request.
+     */
+    public void setMethod(String method) {
```

Comment:
Based on Anu's suggestion to parse the interface lazily, I don't think so. I'm going to be changing this class pretty drastically soon, so I wouldn't think about it too much.